crates/oci/src/client.rs

```diff
@@ -136,14 +148,15 @@ impl Client {
         locked: LockedApp,
         reference: impl AsRef<str>,
         annotations: Option<BTreeMap<String, String>>,
+        predefined_annotations: PredefinedAnnotations,
```

Comment:
The fact that `annotations` is an option of a map and `predefined_annotations` is an enum is somewhat confusing here; how much of this behavior is an external consumer of this API supposed to understand?

Reply:
@radu-matei I am gazing at this and struggling to think of a better design.  I don't want to put it on consumers to construct the predefined annotations, so it shouldn't be a map.  So I am having difficulty getting away from the "I want the predefined annotations or I don't" flag.  I would value suggestions.  Or is it as simple as "we should rename the flag to `infer_predefined_annotations` or something, so it sounds like an action instead of a set" perhaps?

Reply:
@radu-matei I tried some renaming - if that doesn't improve matters then I'd value suggestions - thanks!